chains/btc/uploader/ipfs.go

```diff
@@ -0,0 +1,81 @@
+package uploader
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"mime/multipart"
+	"net/http"
+
+	"github.com/ChainSafe/sygma-relayer/chains/btc/config"
+)
+
+type Uploader interface {
+	Upload(proposals []map[string]interface{}) (string, error)
+}
+
+type IPFSUploader struct {
+	config config.UploaderConfig
+}
+
+func NewIPFSUploader(config config.UploaderConfig) *IPFSUploader {
+	return &IPFSUploader{config: config}
+}
+
+type IPFSResponse struct {
+	IpfsHash string `json:"IpfsHash"`
+}
+
+func (s *IPFSUploader) Upload(dataToUpload []map[string]interface{}) (string, error) {
+
+	// Convert proposals to JSON
+	jsonData, err := json.Marshal(dataToUpload)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a multipart form file
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", "metadata.json")
+	if err != nil {
+		return "", err
+	}
+	_, err = part.Write(jsonData)
+	if err != nil {
+		return "", err
+	}
+	writer.Close()
+
+	// Create a new request
+	req, err := http.NewRequest("POST", s.config.URL, body)
+	if err != nil {
+		return "", err
+	}
+
+	// Set the headers
+	req.Header.Add("Authorization", "Bearer "+s.config.Token)
```

Comment:
I think you can create a client and add these header in the constructor.

Reply:
The issue with that is that you need for each request, first create NewRequest, and only then you are able to set the headers